packages/cli/test/integration/generators/relation.has-one.integration.js

```diff
@@ -0,0 +1,418 @@
+const {
+  expect,
+  TestSandbox
+} = require('@loopback/testlab');
+const {
+  expectFileToMatchSnapshot
+} = require('../../snapshots');
+
+const generator = path.join(__dirname, '../../../generators/relation');
+const {
+  SANDBOX_FILES,
+  SourceEntries
+} = require('../../fixtures/relation');
+const testUtils = require('../../test-utils');
+
+// Test Sandbox
+const SANDBOX_PATH = path.resolve(__dirname, '..', '.sandbox');
+const MODEL_APP_PATH = 'src/models';
+const CONTROLLER_PATH = 'src/controllers';
+const REPOSITORY_APP_PATH = 'src/repositories';
+const sandbox = new TestSandbox(SANDBOX_PATH);
+
+const sourceFileName = [
+  'order.model.ts',
+  'order-class.model.ts',
+  'order-class-type.model.ts',
+];
+const controllerFileName = [
+  'order-customer.controller.ts',
+  'order-class-customer-class.controller.ts',
+  'order-class-type-customer-class-type.controller.ts',
+];
+const repositoryFileName = [
+  'order.repository.ts',
+  'order-class.repository.ts',
+  'order-class-type.repository.ts',
+];
+
+describe('lb4 relation', function () {
+  // eslint-disable-next-line no-invalid-this
+  this.timeout(50000);
+
+  it("rejects relation when source model doesn't have primary Key", async () => {
+    await sandbox.reset();
+
+    const prompt = {
+      relationType: 'hasMany',
+      sourceModel: 'Nokey',
+      destinationModel: 'Customer',
+    };
+
+    return expect(
+      testUtils
+      .executeGenerator(generator)
+      .inDir(SANDBOX_PATH, () =>
+        testUtils.givenLBProject(SANDBOX_PATH, {
+          additionalFiles: SANDBOX_FILES,
+        }),
+      )
+      .withPrompts(prompt),
+    ).to.be.rejectedWith(/Source model primary key does not exist\./);
+  });
+
+  it("rejects relation when destination model doesn't have primary Key", async () => {
+    await sandbox.reset();
+    const prompt = {
+      relationType: 'hasOne',
+      sourceModel: 'Customer',
+      destinationModel: 'Nokey',
+    };
+
+    return expect(
+      testUtils
+      .executeGenerator(generator)
+      .inDir(SANDBOX_PATH, () =>
+        testUtils.givenLBProject(SANDBOX_PATH, {
+          additionalFiles: SANDBOX_FILES,
+        }),
+      )
+      .withPrompts(prompt),
+    ).to.be.rejectedWith(/Target model primary key does not exist/);
+  });
+
+  it('rejects relation when models does not exist', async () => {
+    await sandbox.reset();
+    const prompt = {
+      relationType: 'hasOne',
+      sourceModel: 'Customer',
+      destinationModel: 'Nokey',
+    };
+
+    return expect(
+      testUtils
+      .executeGenerator(generator)
+      .inDir(SANDBOX_PATH, () =>
+        testUtils.givenLBProject(SANDBOX_PATH, {
+          additionalFiles: [
+            // no model/repository files in this project
+          ],
+        }),
+      )
+      .withPrompts(prompt),
+    ).to.be.rejectedWith(/No models found/);
+  });
+
+  it('updates property decorator when property already exist in the model', async () => {
```

Comment:
ditto, the tests should be similar to hasMany as well. Take this test as an example, it is for belongsTo. 
If `@property() customerId` exists, the decorator will be updated to `@belongsTo` for belongsTo relation. 
But if it's the case for `hasOne`, it should throw cause the property already exists.